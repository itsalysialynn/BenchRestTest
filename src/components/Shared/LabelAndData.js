import React from 'react'
import PropTypes from 'prop-types'
import { Col, Row } from 'react-flexbox-grid'
import { TextMuted } from '../../styles/StyledText'

export default function LabelAndData(props) {
  const noChildren = () => {
    const { children } = props
    return (children === null || typeof children === 'undefined' || children === '')
  }

  return (
    <Row>
      <Col className="hidden-md hidden-lg hidden-xl" sm={12} xs={12}>
        <TextMuted><small>{props.label}</small></TextMuted>
      </Col>
      <Col sm={12} xs={12}>
        {noChildren() ? <span>&mdash;</span> : props.children}
      </Col>
    </Row>
  )
}

LabelAndData.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.node,
}
