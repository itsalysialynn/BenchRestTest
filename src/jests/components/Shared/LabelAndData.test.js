import React from 'react'
import { shallow } from 'enzyme'
import LabelAndData from '../../../components/Shared/LabelAndData'

/* global describe, it, expect, test, jest, test  */

describe('rendering label and data', () => {
  const props = { label: 'Amount' }
  const component = shallow(<LabelAndData {...props} />)
  expect(component.html()).toMatch('Amount')

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})


