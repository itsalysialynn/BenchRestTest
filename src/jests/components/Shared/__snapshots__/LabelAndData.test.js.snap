// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering label and data should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabelAndData
    label="Amount"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Col
          className="hidden-md hidden-lg hidden-xl"
          sm={12}
          xs={12}
        >
          <Styled(styled.div)>
            <small>
              Amount
            </small>
          </Styled(styled.div)>
        </Col>,
        <Col
          sm={12}
          xs={12}
        >
          <span>
            —
          </span>
        </Col>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Styled(styled.div)>
            <small>
              Amount
            </small>
          </Styled(styled.div)>,
          "className": "hidden-md hidden-lg hidden-xl",
          "sm": 12,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <small>
              Amount
            </small>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Amount",
            },
            "ref": null,
            "rendered": "Amount",
            "type": "small",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span>
            —
          </span>,
          "sm": 12,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "—",
          },
          "ref": null,
          "rendered": "—",
          "type": "span",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            className="hidden-md hidden-lg hidden-xl"
            sm={12}
            xs={12}
          >
            <Styled(styled.div)>
              <small>
                Amount
              </small>
            </Styled(styled.div)>
          </Col>,
          <Col
            sm={12}
            xs={12}
          >
            <span>
              —
            </span>
          </Col>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Styled(styled.div)>
              <small>
                Amount
              </small>
            </Styled(styled.div)>,
            "className": "hidden-md hidden-lg hidden-xl",
            "sm": 12,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <small>
                Amount
              </small>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Amount",
              },
              "ref": null,
              "rendered": "Amount",
              "type": "small",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span>
              —
            </span>,
            "sm": 12,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "—",
            },
            "ref": null,
            "rendered": "—",
            "type": "span",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
